src/agent.h

```diff
@@ -68,6 +73,8 @@ class MetadataAgent {
 
   HealthChecker health_checker_;
 
+  std::shared_ptr<prometheus::Registry> registry_;
```

Comment:
As I read the docs of prometheus-cpp, they expect clients to have one `Registry` per metric source, and then add each one as a separate `Collectable`. I would suggest following that pattern.
Also see my comment about adding a `Telemetry` class to encapsulate this behavior. Your decision on whether you want to expose the `Registry` interface to the various parts of the agent, or to have a single `Registry` in the `Telemetry` class and expose the individual metrics instead.

Reply:
I think this comment won't be relevant anymore with opencensus implementation.